test/new-e2e/tests/apm/docker_test.go

```diff
@@ -21,139 +21,99 @@ import (
 	"github.com/stretchr/testify/assert"
 )
 
+var (
+	devMode = flag.Bool("devmode", false, "enable dev mode")
+)
+
 type DockerFakeintakeSuite struct {
 	e2e.BaseSuite[environments.DockerHost]
+	transport transport
 }
 
-func TestDockerFakeintakeSuite(t *testing.T) {
+func dockerSuiteOpts(t *testing.T, tr transport, opts ...awsdocker.ProvisionerOption) []e2e.SuiteOption {
+	if !flag.Parsed() {
+		flag.Parse()
+	}
 	devModeEnv, _ := os.LookupEnv("E2E_DEVMODE")
 	options := []e2e.SuiteOption{
-		e2e.WithProvisioner(awsdocker.Provisioner(awsdocker.WithAgentOptions(
-			dockeragentparams.WithAgentServiceEnvVariable("DD_APM_RECEIVER_SOCKET", pulumi.String("/var/run/datadog/apm.socket")), // Enable the UDS receiver in the trace-agent
-			dockeragentparams.WithAgentServiceEnvVariable("STATSD_URL", pulumi.String("unix:///var/run/datadog/dsd.socket")),      // Optional: UDS is more reliable for statsd metrics
-		))),
+		e2e.WithProvisioner(awsdocker.Provisioner(opts...)),
+		e2e.WithStackName(fmt.Sprintf("apm-docker-suite-%s-%v", tr, os.Getenv("CI_PIPELINE_ID"))),
```

Comment:
Added this because the default stack name caused a collision between test suites and errors in the CI

Reply:
You can fix this also by having one base suite and than one implementation per `Test*`, stack name use the suite name as an ID